resim/metrics/proto/metrics.proto

```diff
@@ -0,0 +1,296 @@
+    HISTOGRAM_METRIC_TYPE        = 6;
+    SCALAR_METRIC_TYPE           = 7;
+}
+
+message MetricId {
+    resim.proto.UUID id = 1;
+}
+
+message JobId {
+    resim.proto.UUID id = 1;
+}
+
+// All metrics  associated with a given job.
+// This is the type output by metrics scripts for validation.
+message JobMetrics {
+    JobId job_id = 1;
+
+    MetricCollection job_level_metrics = 2;
+
+    MetricStatus metrics_status = 3;
+
+    repeated MetricsData metrics_data = 4;
+
+    repeated MetricId metric_ids = 5;
+
+    repeated MetricsDataId metrics_data_ids = 6;
+}
+
+// Top-level generic metric collection within JobMetrics.
+// NB: This is separated out in order to allow metrics that are specifically
+// associated with higher-level concepts like batches, or lower-level constructs
+// like actors, which may (for example) PASS, even though the job-level metrics
+// FAIL.
+message MetricCollection {
+    repeated Metric metrics = 1;
+
+    MetricStatus metrics_status = 2;
+}
+
+// A single metric to be shown on a page (such as the job-level insights page).
+message Metric {
+    MetricId metric_id = 1;
+
+    string name = 2;
+
+    MetricType type = 3;
+
+    string description = 4;
+
+    MetricStatus status = 5;
+
+    optional bool should_display = 6;
+
+    MetricValues metric_values = 7;
+
+    optional bool blocking = 8;
+
+    MetricImportance importance = 9;
+
+    oneof parent_id {
+        JobId job_id = 10;
+    }
+}
+
+message MetricValues {
+    oneof metric_values {
+        DoubleSummaryMetricValues  double_metric_values           = 1;
+        DoubleOverTimeMetricValues double_over_time_metric_values = 2;
+        LinePlotMetricValues       line_plot_metric_values        = 3;
+        BarChartMetricValues       bar_chart_metric_values        = 4;
+        StatesOverTimeMetricValues states_over_time_metric_values = 5;
+        HistogramMetricValues      histogram_metric_values        = 6;
+        ScalarMetricValues         scalar_metric_values           = 7;
+    }
+}
+
+message DoubleFailureDefinition {
+    optional double fails_above = 1;
+    optional double fails_below = 2;
+}
+
+message DoubleSummaryMetricValues {
+    MetricsDataId value_data_id = 1;  // DOUBLE array, optionally indexed
+    MetricsDataId status_data_id =
+        2;  // METRIC_STATUS array, optionally indexed
+
+    oneof index {
+        // Optional select index
+        uint32 array_index = 3;
+
+        // Optional select timestamp from timestamp-indexed data
+        google.protobuf.Timestamp timestamp_index = 4;
+
+        // Optional select actor from UUID-indexed data
+        resim.proto.UUID uuid_index = 5;
+
+        // Optional select string from string-indexed data
+        string string_index = 6;
+    }
+
+    DoubleFailureDefinition failure_definition = 7;
+}
+
+message DoubleOverTimeMetricValues {
+    repeated MetricsDataId doubles_over_time_data_id =
+        1;  // INDEXED_DOUBLE_ARRAY_DATA_TYPE, indexed by TIMESTAMP
```

Comment:
I guess all the comments need to be validated, right? It doesn't look like they are complete, but maybe things like the "statuses_data_ids" below is not?

Reply:
Sorry, a bit confused about this one, could you poss clarify a bit? They do need to be validated at least!

Reply:
I think this is resolved now